Source/WebKit/UIProcess/ios/WKApplicationStateTrackingView.mm

```diff
@@ -56,18 +57,18 @@ - (void)willMoveToWindow:(UIWindow *)newWindow
 
     auto page = [_webViewToTrack _page];
     RELEASE_LOG(ViewState, "%p - WKApplicationStateTrackingView: View with page [%p, pageProxyID=%" PRIu64 "] was removed from a window, _lastObservedStateWasBackground=%d", self, page.get(), page ? page->identifier().toUInt64() : 0, page ? page->lastObservedStateWasBackground() : false);
-    _applicationStateTracker = nullptr;
+    _applicationStateTracker->setWindow(nil);
 }
 
 - (void)didMoveToWindow
 {
-    if (!self._contentView.window || _applicationStateTracker)
+    if (!self._contentView.window)
         return;
 
     auto page = [_webViewToTrack _page];
     bool lastObservedStateWasBackground = page ? page->lastObservedStateWasBackground() : false;
 
-    _applicationStateTracker = makeUnique<WebKit::ApplicationStateTracker>(self, @selector(_applicationDidEnterBackground), @selector(_applicationWillEnterForeground), @selector(_willBeginSnapshotSequence), @selector(_didCompleteSnapshotSequence));
+    _applicationStateTracker->setWindow(self.window);
```

Comment:
Should this be `setWindow(self._contentView.window)`?

Reply:
The original code tracked self.window: `UIWindow *window = [m_view.get() window];`.  So I think it's correct to use `self.window` here. But I admit that I don't know what cases might lead to `self.window != self._contentView.window`.